buddy/buddy/UI/authentication/AuthenticationCoordinator.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  AuthenticationCoordinator.swift
+//  buddy
+//
+//  Created by Laura on 04.03.2023..
+//
+
+import UIKit
+import SwiftUI
+
+final class AuthenticationCoordinator: Coordinator {
+    
+    private var navigationController = UINavigationController()
+    weak var parentCoordinator: RootCoordinator?
+    
+    @MainActor func start() -> UIViewController {
+        return showAuthentication()
+    }
+    
+    @MainActor func showAuthentication() -> UIViewController {
+        let authVM = AuthenticationViewModel(persistenceService: PersistenceService())
+        let authVC = UIHostingController(rootView: AuthenticationView(viewModel: authVM))
+                        
+        authVM.onAuthenticatedGoToMain = { [weak self] in
+            self?.callRoot()
+        }
+        
+        navigationController.setViewControllers([authVC], animated: true)
+        return navigationController
+    }
+    
+    private func callRoot() {
+        guard let rootCoordinator = parentCoordinator else { fatalError("Parent coordinator is missing.") }
+        
+        let parentVC = rootCoordinator.start()
+
+        parentVC.modalPresentationStyle = .fullScreen
+        self.navigationController.present(parentVC, animated: true)
```

Comment:
```suggestion
        // same as previous, we dont need this after calling showAsRoot(). Btw this is the point where the memory leak starts, you just keep adding Coordinators and their ViewControllers/ViewModels to the NavigationController stack, then the next coordinator (ProfileCoordinator) adds AuthenticationCoordinator to its navigation stack when you click sign out. The cycle just repeats, and this is why we need showAsRoot() in these cases where we are **"switching" the flow completely**, otherwise it would be okay to just handle flows with navigationController from the coordinator (e.g. if MapCoordinator had some settings flow, you would just push the MapSettingVC on to the navigationController stack)
```